plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TypeConverter.java

```diff
@@ -112,7 +116,18 @@ public static Type toTrinoType(org.apache.iceberg.types.Type type, TypeManager t
         throw new UnsupportedOperationException(format("Cannot convert from Iceberg type '%s' (%s) to Trino type", type, type.typeId()));
     }
 
-    public static org.apache.iceberg.types.Type toIcebergType(Type type)
+    public static org.apache.iceberg.types.Type toIcebergTypeForNewColumn(Type type, AtomicInteger nextFieldId)
+    {
+        org.apache.iceberg.types.Type icebergSchema = toIcebergTypeInternal(type, Optional.empty());
```

Comment:
Rather than calling `assignFreshIds` after, I think this should pass the nextFieldId int through toIcebergTypeInterval